File: eprofiler-core/src/main/java/cz/encircled/eprofiler/Timer.java
package cz.encircled.eprofiler;

/**
 * @author Vlad on 23-May-16.
 */
public class Timer extends Thread {

    public static volatile long now;
    public boolean interrupted = false;

    @Override
    public void run() {
        while (!interrupted) {
            now = System.currentTimeMillis();
            try {
                sleep(1);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}

File: eprofiler-core/src/main/java/cz/encircled/eprofiler/Timer.kt
package cz.encircled.eprofiler

/**
 * @author Vlad on 23-May-16.
 */
class Timer : Thread() {

    override fun run() {
        while (true) {
            now = System.currentTimeMillis()
            try {
                Thread.sleep(1)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

        }
    }

    companion object {

        @Volatile var now: Long = 0
    }
}
